src/mesh/mesh-adapter.factory.ts

```diff
@@ -37,12 +39,14 @@ export class MeshAdapterFactory {
       shellRunner = new LoaderShellRunner(shellRunner, new TTY());
     }
     const timeWindowCalc = new TimeWindowCalculator();
-    const adapters = [];
+    const adapters: MeshAdapter[] = [];
 
     if (this.config.connected.AWS) {
       const aws = new AwsCliFacade(shellRunner);
       const awsAdapter = new AwsMeshAdapter(aws);
-      adapters.push(awsAdapter);
+      const stats = new StatsMeshAdapterDecorator(awsAdapter, MeshPlatform.AWS);
```

Comment:
this one should get the StatsObject in its ctor to avoid the change in call signature. And then use the gathered data in the display stage.